app.rb

```diff
@@ -40,82 +47,121 @@ def create_person
-    @person << student unless student.nil?
-    puts 'Student created successfully!'
+    age = prompt_input('Age: ').to_i
+    name = prompt_input('Name: ')
+    parent_permission = prompt_yes_no('Has parent permission?')
+
+    student = Student.new(age, name: name, parent_permission: parent_permission)
+    @people << student unless student.nil?
+    save_people_data
+    puts 'Student created succesfully!'
   end
 
   def create_teacher
-    print 'Age: '
-    age = gets.chomp.to_i
-    print 'Name: '
-    name = gets.chomp
-    print 'Specialization: '
-    specialization = gets.chomp
-
-    teacher = Teacher.new(specialization, age, name)
-    @person << teacher unless teacher.nil?
+    age = prompt_input('Age: ').to_i
+    name = prompt_input('Name: ')
+    specialization = prompt_input('Specialization: ')
+
+    teacher = Teacher.new(age, specialization, name: name)
+    @people << teacher unless teacher.nil?
+    save_people_data
     puts 'Teacher created successfully!'
   end
 
   def create_book
-    print 'Title: '
-    title = gets.chomp
-
-    print 'Author: '
-    author = gets.chomp
-
+    title = prompt_input('Title: ')
+    author = prompt_input('Author: ')
     @books << Book.new(title, author)
-    puts 'Book created'
+    save_books_data
+    puts 'Book created successfully!'
   end
 
   def create_rental
-    puts 'Select a book from the following list by number: '
-    @books.each_with_index do |book, i|
-      puts "#{i}) Title: #{book.title}, Author: #{book.author}"
-    end
-    book_num = gets.chomp.to_i.to_i
-
-    puts 'Select a person from the following list by number (not id)'
-    @person.each_with_index do |each, i|
-      puts "#{i}) [#{each.class}] Name: #{each.name}, ID: #{each.id}, Age: #{each.age}"
+    puts 'Select a book from the following list by number'
+    @books.each.with_index { |book, idx| puts "#{idx}) Book #{book.title} by #{book.author}" }
+    book_index = gets.chomp.to_i
+    puts 'Select a person from the following list by number (not ID)'
+    @people.each.with_index do |person, idx|
+      puts "#{idx}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
     end
-    person_num = gets.chomp.to_i.to_i
-
-    print 'Date: '
-    date = gets.chomp.to_i
-
-    @rents << Rental.new(date, @person[person_num], @books[book_num])
-    puts 'Rental created successfully '
+    person_index = gets.chomp.to_i
+    date = prompt_input('Date[YYYY/Mon/Date]: ')
+    rental = Rental.new(date, @books[book_index], @people[person_index])
+    @rentals << rental unless rental.nil?
+    save_rentals_data
+    puts 'Rental created successfully!'
   end
 
-  def list_all_person_rentals
-    puts 'List all rents for a given person id'
-    person_id = gets.chomp.to_i
-    list_rentals(person_id)
+  def list_person_rentals_prompt
+    person_id = prompt_input('Enter person ID: ').to_i
+    list_person_rentals(person_id)
   end
```

Comment:
Same here, `list_all_person_rentals` is changed to `list_person_rentals_prompt` yet you are still calling `list_all_person_rentals` to list rental. Please make sure that you update all method call using the correct method name.

This is causing the following error.
![image](https://github.com/GraceHtet/school_library/assets/38088818/0c17ef89-7cc6-4802-b73d-d674bdb6e913)


## Positives

I can see that all data is saved correctly. Great job! 💯 
Just make sure that the saved data can be listed for people, books, and rentals to fully take advantage of the preserved data functionalities.

Reply:
well resolved! ✔️